src/compile/common.ts

```diff
@@ -101,20 +92,27 @@ export function formatSignalRef(
       )
     };
   } else {
-    return {
-      signal: `''+${vgField(fieldDef, {expr})}`
-    };
+    const format = numberFormat(fieldDef, specifiedFormat, config);
+    if (isBinning(fieldDef.bin)) {
+      const startField = vgField(fieldDef, {expr});
+      const endField = vgField(fieldDef, {expr, binSuffix: 'end'});
+      return {
+        signal: binFormatExpression(startField, endField, format, config)
+      };
+    } else if (fieldDef.type === 'quantitative') {
+      return {
+        signal: `${formatExpr(vgField(fieldDef, {expr, binSuffix: 'range'}), format)}`
+      };
+    } else {
+      return {signal: `''+${vgField(fieldDef, {expr})}`};
```

Comment:
Why the `''+` here?

Reply:
To convert to string first?

Reply:
But why?

Reply:
I just tested `return {signal: `${vgField(fieldDef, {expr})}`};`, and use number field, and it works fine. So, I'm not sure why do we need `''+`. Do you want me to remove it?

Reply:
Please think through it but remove it if it's not needed.

Reply:
<img width="864" alt="Screen Shot 2019-03-24 at 21 25 14" src="https://user-images.githubusercontent.com/30903997/54895135-5f666200-4e7b-11e9-8840-6d6bd960915f.png">

It will convert `null` to string. The left one is with `''+` and the right one is without. So, I think we should keep it.

Reply:
Yep 👍. Thanks for checking.